### PR TITLE
Update DateRange.tsx

### PR DESCRIPTION
**Pull request by Aryam2121**
timeStep Prop: Control the rounding by passing a timeStep prop (e.g., 15 for 15-minute intervals).
Testing: Implement unit tests outside of this file to validate the functionality and edge cases of rounding behavior.
UI Considerations: The layout remains responsive and adjusts to the presence or absence of withTime.solve issue #1452  and old pull request changes review #2034

### CODE REVIEW

File: frontend/taipy-gui/src/components/Taipy/DateRange.tsx [hunk 1]
```diff
@@ -40,40 +40,26 @@ interface DateRangeProps extends TaipyActiveProps, TaipyChangeProps {
...
-        }
-    } else {
-        dates = json as string[];
-    }
-    if (Array.isArray(dates) && dates.length) {
-        return dates.map((d) => getDateTime(d, tz, withTime)) as [Date, Date];
-    }
-    return [null, null];
+// Function to round the time based on the given step
+const roundToStep = (date: Date | null, step: number | undefined): Date | null => {
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/DateRange.tsx:49]: Should go in a dateUtils.ts so that we can reuse it in dateSelector.tsx and timeSelector.tsx ...

File: frontend/taipy-gui/src/components/Taipy/DateRange.tsx [hunk 2]
```diff
@@ -40,40 +40,26 @@ interface DateRangeProps extends TaipyActiveProps, TaipyChangeProps {
...
+    if (!date || !step) return date;
+    const roundedMinutes = Math.ceil(date.getMinutes() / step) * step;
+    return setMinutes(setHours(date, date.getHours()), roundedMinutes);
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/DateRange.tsx:52]: what happens if step = 40 and current min = 50 ?
roundedMinutes would be 80 
How does DateFns handles this ?

File: frontend/taipy-gui/src/components/Taipy/DateRange.tsx [hunk 3]
```diff
@@ -231,7 +218,6 @@ const DateRange = (props: DateRangeProps) => {
                             />
                         </>
                     )}
-                    {props.children}
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/DateRange.tsx:234]: Why ?

File: frontend/taipy-gui/src/components/Taipy/DateRange.tsx [hunk 4]
```diff
@@ -40,40 +40,26 @@ interface DateRangeProps extends TaipyActiveProps, TaipyChangeProps {
     labelEnd?: string;
     separator?: string;
     width?: string | number;
+    timeStep?: number; // New prop to define time step rounding (e.g., 15 for 15 minutes)
 }
 
 const textFieldProps = { textField: { margin: "dense" } } as BaseDateTimePickerSlotProps<Date>;
 
-const getRangeDateTime = (
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/DateRange.tsx:47]: Why ?

File: frontend/taipy-gui/src/components/Taipy/DateRange.tsx [hunk 5]
```diff
@@ -40,40 +40,26 @@ interface DateRangeProps extends TaipyActiveProps, TaipyChangeProps {
...
 const DateRange = (props: DateRangeProps) => {
-    const { updateVarName, withTime = false, id, propagate = true, separator = "-" } = props;
+    const { updateVarName, withTime = false, id, propagate = true, separator = "-", timeStep } = props;
     const dispatch = useDispatch();
     const formatConfig = useFormatConfig();
     const tz = formatConfig.timeZone;
     const [value, setValue] = useState<[Date | null, Date | null]>([null, null]);
     const [startProps, setStartProps] = useState<DateProps>({});
     const [endProps, setEndProps] = useState<DateProps>({});
-    const module = useModule();
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/DateRange.tsx:76]: Why ?

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
this is not finished I guess ?

**Comment by FredLL-Avaiga** [comment]
Are you still working on this @Aryam2121 ?
if you do, update your branch, resolve the conflicts and answer the questions, tx.
if not please close the PR.

**Comment by FredLL-Avaiga** [comment]
> Are you still working on this @Aryam2121 ? if you do, update your branch, resolve the conflicts and answer the questions, tx. if not please close the PR.

WIthout a reply from you @Aryam2121, we'll have to close this PR ...

**Comment by Aryam2121** [comment]
I am updated already

**Comment by FredLL-Avaiga** [comment]
can you resolve the conflicts and fix the failing checks ?

**Comment by FredLL-Avaiga** [comment]
no activity, closing
